wandb/apis/public/utils.py

```diff
@@ -33,3 +36,33 @@ def parse_s3_url_to_s3_uri(url) -> str:
     s3_uri = f"s3://{bucket_name}/{key}"
 
     return s3_uri
+
+
+class PathType(Enum):
+    """We have lots of different paths users pass in to fetch artifacts, projects, etc.
+
+    This enum is used for specifying what format the path is in given a string path.
+    """
+
+    PROJECT = "PROJECT"
+    ARTIFACT = "ARTIFACT"
+
+
+def parse_org_from_registry_path(path: str, path_type: PathType) -> str:
+    """Parse the org from a registry path.
+
+    Essentially fetching the "entity" from the path but for Registries the entity is actually the org.
+
+    Args:
+        path (str): The path to parse. Can be a project path <entity>/<project> or <project> or an
+        artifact path like <entity>/<project>/<artifact> or <project>/<artifact> or <artifact>
+        path_type (PathType): The type of path to parse.
+    """
+    parts = path.split("/")
+    expected_parts = 3 if path_type == PathType.ARTIFACT else 2
+
+    if len(parts) >= expected_parts:
+        org, project = parts[:2]
+        if is_artifact_registry_project(project):
+            return org
+    return ""
```

Comment:
probably better to put it somewhere closer to artifacts it is literally only used for artifacts